### PR TITLE
Fix the java generation pipeline and improve fetch APIs

### PR DESCRIPTION
**Pull request by andreaTP**
This should be a fix for this failing pipeline:
https://github.com/redhat-developer/app-services-sdk-core/actions/runs/4293215594/jobs/7480614471

Since I was touching it, I changed the `fetch_api` script from Python to Bash following the [principle of least power](https://en.wikipedia.org/wiki/Rule_of_least_power), and given that most of the generation logic is already handled in bash in this repo.

### CODE REVIEW

File: scripts/fetch-api.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "https://sso.redhat.com/auth/realms/redhat-external/apis/openapi.yaml"
+)
+
+for endpoint in ${endpoints[*]}; do
+    filename=$(echo "${endpoint##*/}")
+
+    if [[ ${filename} == "openapi.yaml" ]]; then
+        filename="service-accounts.yaml"
+    fi
+    if [[ ${filename} == "openapi.json" ]]; then
```
Comment (jsenko) [inline scripts/fetch-api.sh:22]: The name should probably be `openapi-gen.json` like so:
```
https://raw.githubusercontent.com/Apicurio/apicurio-registry/main/app/src/main/resources-unfiltered/META-INF/resources/api-specifications/registry/v2/openapi-gen.json
```

Comment (andreaTP) [reply scripts/fetch-api.sh:22]: I have used a different file:
`https://raw.githubusercontent.com/Apicurio/apicurio-registry/main/common/src/main/resources/META-INF/openapi.json`

this is the source of truth for the Server Side codegen, any specific reason for using the alternative one?

Comment (jsenko) [reply scripts/fetch-api.sh:22]: The `*-gen` is specifically modified for the Openapi generator to generaate stuff we need in the client.

Comment (andreaTP) [reply scripts/fetch-api.sh]: Agreed, done here: https://github.com/redhat-developer/app-services-sdk-core/pull/98/commits/2c578b5bc193c8d9668fc234c52337424b184f3b 👍

**Review by dimakis — APPROVED** [review]
Lgtm.

File: scripts/generate-java.sh [hunk 2]
```diff
@@ -159,4 +159,4 @@ then
         --ignore-file-override=.openapi-generator-ignore
 fi
 
-rm .openapi/*.processed
\ No newline at end of file
+rm -Rf .openapi/*.processed
```
Comment (dimakis) [inline scripts/generate-java.sh:162]: Great catch 😁

**Review by rkpattnaik780 — APPROVED** [review]

**Pull request merged**